### PR TITLE
[WebLink] Added missing types

### PR DESCRIPTION
**Pull request by tarlepp**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | -
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Review by dunglas — APPROVED** [review]

File: src/Symfony/Component/WebLink/Link.php [hunk 1]
```diff
@@ -134,7 +134,7 @@ public function withoutRel($rel)
      *
      * @return static
      */
-    public function withAttribute($attribute, $value)
+    public function withAttribute(string $attribute, string|\Stringable|int|float|bool|array $value)
     {
         $that = clone $this;
         $that->attributes[$attribute] = $value;
```
Comment (Nyholm) [inline src/Symfony/Component/WebLink/Link.php:140]: Lets update the type for the `$attribute` property too. It currently says it should be an array of strings.

**Comment by Nyholm** [comment]
Thank you. 

I see this error on Travis: 

> Declaration of Symfony\Component\WebLink\GenericLinkProvider::getLinksByRel(string $rel): array must be compatible with Psr\Link\LinkProviderInterface::getLinksByRel($rel) in /home/travis/build/symfony/symfony/src/Symfony/Component/WebLink/GenericLinkProvider.php on line 49

It is the same issue as this: https://3v4l.org/uM5S9

Let's update `psr/link` to support only ^1.1. Both the root composer.json and the composer.json in the WebLink component. Travis should be happy then.

**Comment by Nyholm** [comment]
Thank you. 

I see this error on Travis: 

> Declaration of Symfony\Component\WebLink\GenericLinkProvider::getLinksByRel(string $rel): array must be compatible with Psr\Link\LinkProviderInterface::getLinksByRel($rel) in /home/travis/build/symfony/symfony/src/Symfony/Component/WebLink/GenericLinkProvider.php on line 49

It is the same issue as this: https://3v4l.org/uM5S9

Let's update `psr/link` to support only ^1.1. Both the root composer.json and the composer.json in the WebLink component. Travis should be happy then.

**Comment by Nyholm** [comment]
Thank you. 

I see this error on Travis: 

> Declaration of Symfony\Component\WebLink\GenericLinkProvider::getLinksByRel(string $rel): array must be compatible with Psr\Link\LinkProviderInterface::getLinksByRel($rel) in /home/travis/build/symfony/symfony/src/Symfony/Component/WebLink/GenericLinkProvider.php on line 49

It is the same issue as this: https://3v4l.org/uM5S9

Let's update `psr/link` to support only ^1.1. Both the root composer.json and the composer.json in the WebLink component. Travis should be happy then.

**Comment by Nyholm** [comment]
Thank you. 

I see this error on Travis: 

> Declaration of Symfony\Component\WebLink\GenericLinkProvider::getLinksByRel(string $rel): array must be compatible with Psr\Link\LinkProviderInterface::getLinksByRel($rel) in /home/travis/build/symfony/symfony/src/Symfony/Component/WebLink/GenericLinkProvider.php on line 49

It is the same issue as this: https://3v4l.org/uM5S9

Let's update `psr/link` to support only ^1.1. Both the root composer.json and the composer.json in the WebLink component. Travis should be happy then.

**Comment by tarlepp** [comment]
> Let's update `psr/link` to support only ^1.1. Both the root composer.json and the composer.json in the WebLink component. Travis should be happy then.

Done

File: src/Symfony/Component/WebLink/composer.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
     },
     "require": {
         "php": ">=8.0.2",
-        "psr/link": "^1.0"
+        "psr/link": "^1.1"
```
Comment (derrabus) [inline src/Symfony/Component/WebLink/composer.json:23]: Shall we add a conflict rule for `ext-psr`? If we bump here, the component will break if (any released version of) `ext-psr` is installed.

Comment (nicolas-grekas) [reply src/Symfony/Component/WebLink/composer.json:23]: I wouldn't care, that's not our problem

**Comment by nicolas-grekas** [comment]
Thank you @tarlepp.

**Pull request merged**